datafusion/functions-nested/src/map.rs

```diff
@@ -84,22 +80,42 @@ fn make_map_batch(args: &[ColumnarValue]) -> Result<ColumnarValue> {
 
             row_keys
                 .iter()
-                .try_for_each(|key| check_unique_keys(key.as_ref()))?;
+                .try_for_each(|key| validate_map_keys(key.as_ref()))?;
         }
         ColumnarValue::Scalar(_) => {
-            check_unique_keys(key_array)?;
+            validate_map_keys(key_array)?;
         }
     }
 
     let values = get_first_array_ref(values_arg)?;
+
+    // For const evaluation with NULL maps, we must use make_map_array_internal
+    // because make_map_batch_internal doesn't handle NULL list elements correctly
+    if can_evaluate_to_const && keys.null_count() > 0 {
+        // If there are NULL maps, use the array path which handles them correctly
+        return if let DataType::LargeList(..) = keys_arg.data_type() {
+            make_map_array_internal::<i64>(keys, values)
+        } else {
+            make_map_array_internal::<i32>(keys, values)
+        };
+    }
```

Comment:
We should combine it with the existing check inside `make_map_batch_internal`:

https://github.com/apache/datafusion/blob/becc71be04732ca2f62139192dcc76caf6c08fd3/datafusion/functions-nested/src/map.rs#L133-L139

e.g. `!can_evaluate_to_const || keys.null_count() > 0`

Though it would be nice if we could fix the scalar fast path in `make_map_batch_internal` to not need this workaround

(Also this made me realize FixedSizeLists are broken for key array input, since it handles on List/LargeList)